// @vercel/hono@46.1.1/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  build: () => build,
  findEntrypoint: () => findEntrypoint,
  shouldServe: () => shouldServe,
  startDevServer: () => startDevServer,
  version: () => version
});
module.exports = __toCommonJS(src_exports);

// src/build.ts
var import_node = require("@vercel/node");
var import_path = require("path");
var import_fs = __toESM(require("fs"));
var REGEX = /(?:from|require|import)\s*(?:\(\s*)?["']hono["']\s*(?:\))?/g;
var build = async (args) => {
  const entrypoint = findEntrypoint(args.files);
  process.env.EXPERIMENTAL_NODE_TYPESCRIPT_ERRORS = "1";
  return (0, import_node.build)({
    ...args,
    entrypoint,
    considerBuildCommand: true,
    entrypointCallback: (preparedFiles) => {
      return findEntrypoint(preparedFiles);
    }
  });
};
var findEntrypoint = (files) => {
  const validFilenames = [
    ["app"],
    ["index"],
    ["server"],
    ["src", "app"],
    ["src", "index"],
    ["src", "server"]
  ];
  const validExtensions = ["js", "cjs", "mjs", "ts", "cts", "mts"];
  const validEntrypoints = validFilenames.flatMap(
    (filename) => validExtensions.map((extension) => `${filename.join(import_path.sep)}.${extension}`)
  );
  const entrypoints = validEntrypoints.filter((entrypoint2) => {
    const matches = files[entrypoint2] !== void 0;
    if (matches) {
      const file = files[entrypoint2];
      if (file.type === "FileBlob") {
        const content = file.data.toString();
        const matchesContent = content.match(REGEX);
        return matchesContent !== null;
      }
      if (file.type === "FileFsRef") {
        const content = import_fs.default.readFileSync(file.fsPath, "utf-8");
        const matchesContent = content.match(REGEX);
        return matchesContent !== null;
      }
    }
    return false;
  });
  const entrypoint = entrypoints[0];
  if (entrypoints.length > 1) {
    console.warn(
      `Multiple entrypoints found: ${entrypoints.join(", ")}. Using ${entrypoint}.`
    );
  }
  if (!entrypoint) {
    const entrypointsForMessage = validFilenames.map((filename) => `- ${filename.join(import_path.sep)}.{${validExtensions.join(",")}}`).join("\n");
    throw new Error(
      `No valid entrypoint found. Valid entrypoints are:
${entrypointsForMessage}`
    );
  }
  return entrypoint;
};

// src/index.ts
var import_node2 = require("@vercel/node");
var version = 3;
var shouldServe = async (opts) => {
  const requestPath = opts.requestPath.replace(/\/$/, "");
  if (requestPath.startsWith("api") && opts.hasMatched) {
    return false;
  }
  return true;
};
var startDevServer = async (opts) => {
  const entrypoint = findEntrypoint(opts.files);
  process.env.EXPERIMENTAL_NODE_TYPESCRIPT_ERRORS = "1";
  return (0, import_node2.startDevServer)({
    ...opts,
    entrypoint
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  build,
  findEntrypoint,
  shouldServe,
  startDevServer,
  version
});
